insights/core/__init__.py

```diff
@@ -689,10 +689,15 @@ class YAMLParser(Parser, LegacyItemAccess):
     """
     def parse_content(self, content):
         try:
+            try:
```

Comment:
If the import in the try body fails, an `ImportError` will need to raise every time through parse_content to get the python loader out of the module cache in the except block. Let's move the try/except of the import to the top of the module so we only pay the cost once.